core/print_api.php

```diff
@@ -2006,6 +2035,46 @@ function print_bug_attachment_preview_image( array $p_attachment ) {
 	echo '</a></div>';
 }
 
+/**
+ * Prints the preview of an audio file attachment.
+ * @param array $p_attachment An attachment array from within the array returned by the file_get_visible_attachments() function.
+ * @param string $p_file_type mime type
+ * @param boolean $p_preload true to preload audio, false otherwise.
+ * @return void
+ */
+function print_bug_attachment_preview_audio( array $p_attachment, $p_file_type, $p_preload ) {
+	$t_file_url = $p_attachment['download_url'] . '&show_inline=1' . form_security_param( 'file_show_inline' );
+	$t_preload = $p_preload ? '' : ' preload="none"';
+
+	echo "\n<div class=\"bug-attachment-preview-audio\">";
+	echo '<a href="' . string_attribute( $p_attachment['download_url'] ) . '">';
+	echo '<audio controls="controls"' . $t_preload . '>';
+	echo '<source src="' . string_attribute( $t_file_url ) . '" type="' . string_attribute( $p_file_type ) . '">';
+  	echo lang_get( 'browser_does_not_support_audio' );
+	echo "</audio>";
+	echo "</a></div>";
+}
+
+/**
+ * Prints the preview of an video file attachment.
+ * @param array $p_attachment An attachment array from within the array returned by the file_get_visible_attachments() function.
+ * @param string $p_file_type mime type
+ * @param boolean $p_preload true to preload video, false otherwise.
+ * @return void
+ */
+function print_bug_attachment_preview_video( array $p_attachment, $p_file_type, $p_preload ) {
```

Comment:
This is pretty much the same code as *print_bug_attachment_preview_audio()* the only difference is `s/audio/video/`, wouldn't it make sense to have a generic function ?